event_management/src/main/java/com/example/event_management/service/UserEventService.java

```diff
@@ -0,0 +1,109 @@
+package com.example.event_management.service;
+
+import com.example.event_management.model.UserEvent;
+import com.example.event_management.repository.UserEventRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+@Service
+public class UserEventService {
+
+    @Autowired
+    UserEventRepository userEventRepository;
+
+    public void deleteByEventId(String id) {
+        List<UserEvent> userEventList = userEventRepository.findAll();
+
+        for (UserEvent userEvent : userEventList) {
+            if (userEvent.getEventId() != null && userEvent.getEventId().toString().equals(id)) {
+                userEventRepository.deleteById(userEvent.getId());
+            }
+        }
+    }
+
+    public void deleteByUserId(String id) {
+        List<UserEvent> userEventList = userEventRepository.findAll();
+
+        for (UserEvent userEvent : userEventList) {
+            if (userEvent.getUserId() != null && userEvent.getUserId().toString().equals(id)) {
+                userEventRepository.deleteById(userEvent.getId());
+            }
+        }
+    }
+
+    public void bookAnEvent(UserEvent userEvent) {
+
+        UserEvent bookEvent = userEventRepository.findByEventIdAndUserId(userEvent.getEventId(), userEvent.getUserId());
+
+        if(Objects.nonNull(bookEvent)){
+
+            if(bookEvent.isBooked()) {
+                bookEvent.setBooked(false);
+            }else{
+                bookEvent.setBooked(true);
+            }
```

Comment:
Take booked status in request
```suggestion
            bookEvent.setBooked(userEvent.getBooked())
```

Reply:
updated and added increment and decrement of seatsLeft after booking